cli/injector/cpu_pressure.go

```diff
@@ -18,10 +21,40 @@ var cpuPressureCmd = &cobra.Command{
 	PreRun: func(cmd *cobra.Command, args []string) {
 		// prepare spec
 		spec := v1beta1.CPUPressureSpec{}
+		podCores, err := getPodCores()
+		if err != nil {
+			log.Fatalw("error creating the node injector", "error", err)
+			return
+		}
+
+		cores, err := cpuset.Parse(podCores)
+		if err != nil {
+			log.Fatalw("error creating the node injector", "error", err)
+			return
+		}
 
+		log.Infow(fmt.Sprintf("target identified to be running on %d cores", cores.Size()), "cores", cores.ToSlice())
 		// create injector
 		for _, config := range configs {
-			injectors = append(injectors, injector.NewCPUPressureInjector(spec, injector.CPUPressureInjectorConfig{Config: config}))
+			injectors = append(injectors, injector.NewCPUPressureInjector(spec, injector.CPUPressureInjectorConfig{Config: config, DistinctPodCores: cores}))
 		}
 	},
 }
+
+func getPodCores() (string, error) {
```

Comment:
I'm not sure it should be part of the `command` file 🤔 (that is handling inputs/outputs, but should not necessarily contains business/disruption logic IMO)

It may be more the `injector/cpu_pressure.go` responsibility to detect what are the amount of cores.

It relates to above loop, migrating it to a single call having all config could be an option, there are probably several others.

Reply:
I agree. It may require some extra changes but this logic should probably be done in the injector package since it is where the injection logic is expected to be (especially to handle errors properly later as I said in another comment).

Reply:
there's some tradeoff here. 

Essentially I need to know the `[]cores` from all the `configs`. Now, an injector ideally shouldn't have to care about another injector (hence I didn't pass `configs` to `NewCPUPressureInjector`). 

On the separation of concern aspect, the `command` in this PR is only extracting the number of cores. It hasn't involved further injection logic. It prepares the number of cores and pass to a concrete injector to actually inject.

Let's discuss more

Reply:
One pattern would be to delegate that to the stresser itself. Right now, we're creating one stresser per config == per container. Stressers are independent, they have no idea about other stressers. 
We could redefine that by having a stresser manager to which you can pass additional cpuset over time. The stresser manager would spin one stresser for each given cpuset and update GOMAXPROCS accordingly. It would delegate the logic of keeping track of cpusets in another package which can be called by multiple injectors and have a glocal array of cpusets.

Let me know if it is not clear, I can give an example.